ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTrigger.java

```diff
@@ -13,26 +13,63 @@
 
 package tech.pegasys.teku.statetransition.forkchoice;
 
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import tech.pegasys.teku.ethereum.performance.trackers.BlockProductionPerformance;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.time.TimeProvider;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 
 public class ForkChoiceTrigger {
-
+  public static final int WARNING_TIME_MILLIS = 250;
+  public static final int DEBUG_TIME_MILLIS = 10;
+  private static final Logger LOG = LogManager.getLogger();
   private final ForkChoiceRatchet forkChoiceRatchet;
   private final ForkChoice forkChoice;
+  private final int attestationWaitLimitMillis;
+  private final TimeProvider timeProvider;
 
-  public ForkChoiceTrigger(final ForkChoice forkChoice) {
+  public ForkChoiceTrigger(
+      final ForkChoice forkChoice,
+      final int attestationWaitLimitMillis,
+      final TimeProvider timeProvider) {
     this.forkChoiceRatchet = new ForkChoiceRatchet(forkChoice);
     this.forkChoice = forkChoice;
+    this.attestationWaitLimitMillis = attestationWaitLimitMillis;
+    this.timeProvider = timeProvider;
   }
 
   public void onSlotStartedWhileSyncing(final UInt64 nodeSlot) {
     forkChoiceRatchet.ensureForkChoiceCompleteForSlot(nodeSlot).join();
   }
 
   public void onAttestationsDueForSlot(final UInt64 nodeSlot) {
-    forkChoiceRatchet.ensureForkChoiceCompleteForSlot(nodeSlot).join();
+    try {
+      final UInt64 startTimeMillis = timeProvider.getTimeInMillis();
+      forkChoiceRatchet
+          .ensureForkChoiceCompleteForSlot(nodeSlot)
+          .get(attestationWaitLimitMillis, TimeUnit.MILLISECONDS);
+      final UInt64 duration = timeProvider.getTimeInMillis().minusMinZero(startTimeMillis);
+      if (duration.isGreaterThanOrEqualTo(WARNING_TIME_MILLIS)) {
+        LOG.warn(
+            "Took {} ms waiting for fork choice to complete at slot {}, when attestations were due.",
+            duration,
+            nodeSlot);
+      } else if (duration.isGreaterThanOrEqualTo(DEBUG_TIME_MILLIS)) {
+        LOG.debug(
+            "Took {} ms waiting for fork choice to complete at slot {}, when attestations were due.",
+            duration,
+            nodeSlot);
+      }
+    } catch (InterruptedException | TimeoutException e) {
+      LOG.error("Failed to wait for fork choice to complete for slot " + nodeSlot, e);
```

Comment:
NIT: could add the value of `attestationWaitLimitMillis` to the message so we know how long we waited for w/o having to check the config on that node.

Reply:
it'd be in every message... i can add it if you think it wont just be noise... mostly i figured slot would be most informative

Reply:
I guess it doesn't matter too much if we aren't planning to play too much with the time limit config. We can assume it will be 1.5s most of the time.

I was thinking something similar to what we have for slow db operations:
```
DB operation KvStoreDatabase::doUpdate took too long: 1762 ms. The alert threshold is set to: 1000 ms.
```